### DISCUSSION TITLE
Composition API

### DISCUSSION THREAD

**Discussion by loxy** [taiga-family/taiga-ui#4037] [Q&A] [answered]
Hi again!

There is this new [Composition API in v15](https://angular.io/guide/directive-composition-api) which is quite cool. Would be great if we could write our own button directive on top of yours ❤️ for example!

Currently it is not possible to use it with non standalone directives. 

Are you aware of that? Any plans to change something in that direction? 

Best regards,
Kersten

**Comment by waterplea** [comment]
We cannot bump the Angular version because it's not backwards compatible. When the majority of projects using Taiga UI will migrate to a particular Angular version or higher — then we update. However I'm not sure how this would help you, Composition API is for components authors, not users. You can already apply any amount of directives on top of our buttons.

> The directive composition API lets you apply directives to a component's host element from within the component TypeScript class.

**Comment by waterplea** [comment, marked as answer]
Taiga UI now fully embraces Composition API
